tests/dflydev/tests/util/antPathMatcher/AntPathMatcherTest.php

```diff
@@ -95,7 +95,12 @@ public function testMatchesProvider()
                 'com/foo/',
                 array('com/foo/bar.jsp','com/foo/bar/baz.jsp',),
                 array('com.txt', 'com/foo.txt'),
-            )
+            ),
+            array(
+                'com/**',
```

Comment:
I suspect this could actually be a "bug" in Sculpin. Specifically, in `AntPathFilter`:

https://github.com/sculpin/sculpin/blob/5f705d845b2dc980ed91b79c49ccaa5f64cbdda0/src/Sculpin/Core/Source/Filter/AntPathFilter.php#L38

My assumption would have been that the path should have been `posts/**` and `posts2/**`. However, I'm guessing whatever `isPattern()` does with the information provided ("posts" and "posts2") decides to use them raw vs appending `/**` like expected.

If you still have the time and energy, could you verify for me what Sculpin is actually doing in `AntPathFilter` to see which patterns it is actually using to pass to the matches?

----

I'm not saying this PR is wrong or won't be merged. I just want to be very certain that it isn't going to have weird side-effects.

In particular, I'd have assumed this test should pass even without rooting the regular expression. If that truly is not the case, then I don't know what to say. :)

Reply:
> I suspect this could actually be a "bug" in Sculpin. Specifically, in AntPathFilter...

`AntPathMatcher::isPattern()` returns `true` if any of the following are satisfied:

 * path contains `*`
 * path contains `?`
 * path ends with `/`

I'll note that `isPattern()` curiously provides the integer offset for `substr()` as a string value (i.e. `'-1'`), but I don't think that's a problem since PHP will coerce it.

Per the Sculpin config I shared in https://github.com/sculpin/sculpin/issues/475, the incoming paths would be "_posts" and "_posts2". Neither of these would be considered a pattern, so Sculpin will append `/**` to each (although I expect a single `/` would work just as well per the "shorthand" discussed above in https://github.com/dflydev/dflydev-util-antPathMatcher/pull/1#discussion_r1083243962).

> If you still have the time and energy, could you verify for me what Sculpin is actually doing in AntPathFilter to see which patterns it is actually using to pass to the matches?

No problem. I actually started my investigation in AntPathFilter, which is what ultimately lead me to this PR. I can confirm that the "_posts" and _"posts2" paths are converted to "_posts/**" and "_posts2/**", respectively. I don't think there's any problem with `AntPathMatcher::isPattern()` or the logic in `AntPathFilter::__construct()`.

> I'm not saying this PR is wrong or won't be merged. I just want to be very certain that it isn't going to have weird side-effects.

Understandable. I'm also cognizant that this code hasn't been touched in over ten years, so there's no pressure on my end to merge this until you're comfortable doing so.

Reply:
I guess I'm curious why `_posts/` and `_posts2/` would ever match the same thing, but that's fine. It clearly wasn't working as expected before, and seems to be now, so that's good enough for me. :)

Reply:
@simensen: I think there was a Markdown formatting issue and some text was inadvertently replaced with formatting. The `_posts/` path was converted to the pattern `_posts/**`, and the lack of regex anchoring meant that `_posts` matched _both_ `posts` and `posts2`.

This meant that the generator for posts picked up content from _both_ directories; however, there was never any problem in the reverse direction (i.e. posts2 picking up content from `posts/`).